src/cpu/dyn_cache.h

```diff
@@ -741,16 +792,26 @@ static void cache_init(bool enable) {
 		if (cache_code_start_ptr==NULL) {
 			// allocate the code cache memory
 #if defined (WIN32)
-			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP,
+			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,cache_code_start_sz,
 				MEM_COMMIT,PAGE_EXECUTE_READWRITE);
 			if (!cache_code_start_ptr)
-				cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+				cache_code_start_ptr=(Bit8u*)malloc(cache_code_start_sz);
 #else
-			cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+			int map_flags = MAP_PRIVATE | MAP_ANON;
+			int prot_flags = PROT_READ | PROT_WRITE | PROT_EXEC;
+#if defined(HAVE_MAPJIT)
+			map_flags |= MAP_JIT;
 #endif
+#if defined(HAVE_MMAP)
+			cache_code_start_ptr=(Bit8u*)mmap(NULL, cache_code_start_sz, prot_flags, map_flags, -1, 0);
+			if (cache_code_start_ptr == MAP_FAILED)
+				E_Exit("Allocating dynamic core cache memory failed with errno %d", errno);
+#else
+			cache_code_start_ptr=(Bit8u*)malloc(cache_code_start_sz);
```

Comment:
Probably also need a `reinterpret_cast<uint8_t*>`.

Reply:
Should it be `reinterpret_cast<Bit8u*>`?

Reply:
Oh; if a static_cast will work, yes - go with that. 
In new or changed code, we always prefer C++11's <cstdint>fixed-size types over the older DOSBox and SDL typedefs.  They severed a purpose prior to <cstdint> but are unnecessary now.  Whenever possible, we want to prefer standards in place of prior work-arounds / macros / typedefs because it's easier for newcomers (small benefits, but they add up 😅 )